environment-mac.yaml

```diff
@@ -0,0 +1,32 @@
+name: ldm
+channels:
+  - apple
+  - conda-forge
+  - pytorch-nightly
+  - defaults
+dependencies:
+  - python=3.10.4
+  - pip=22.1.2
+  - pytorch
+  - torchvision
+  - numpy=1.23.1
+  - pip:
+    - albumentations==0.4.6
+    - opencv-python==4.6.0.66
+    - pudb==2019.2
+    - imageio==2.9.0
+    - imageio-ffmpeg==0.4.2
+    - pytorch-lightning==1.4.2
+    - omegaconf==2.1.1
+    - test-tube>=0.7.5
+    - streamlit==1.12.0
+    - pillow==9.2.0
+    - einops==0.3.0
+    - torch-fidelity==0.3.0
+    - transformers==4.19.2
+    - torchmetrics==0.6.0
+    - kornia==0.6.0
+    - -e git+https://github.com/openai/CLIP.git@main#egg=clip
+    - -e git+https://github.com/CompVis/taming-transformers.git@master#egg=taming-transformers
+    - -e git+https://github.com/lstein/k-diffusion.git@master#egg=k-diffusion
```

Comment:
I recommend to prefer my fork of k-diffusion:  
https://github.com/crowsonkb/k-diffusion/compare/master...lstein:k-diffusion:master

it includes a few MPS-specific fixes:  
- `sigma_to_t`: silence a fallback-to-CPU warning when sorting a tiny matrix
- `t_to_sigma`: use `t-t_floor` insted of `t.frac()` (which is unsupported on MPS)
- `to_d`: support cloning coeff to prevent values' becoming ±inf and image becoming black
- `sample_heun`, `sample_lms`: opt into cloning feature on `to_d`

some fixes for Karras et al noise schedule and samplers, to optimize it for discrete sigmas like those stable diffusion was learned on:  
- `sample_euler`, `sample_heun`, `sample_dpm_2`: time step discretization
- `get_sigmas_karras`: don't include 0 in our noise schedule; SD's discrete sigmas don't go that low.

Reply:
Good to know. I am modifying environment.yaml to point to @Birch-san 's fork rather than mine. I think the only reason I forked was because the setup info was missing upstream.